Tests/EndToEnd.tests.ts

```diff
@@ -273,6 +275,95 @@ describe("EndToEnd", () => {
         });
     });
 
+    describe("W3C mode", () => {
+        var sandbox: sinon.SinonSandbox;
+
+        beforeEach(() => {
+            sandbox = sinon.sandbox.create();
+        });
+
+        afterEach(() => {
+            // Dispose the default app insights client and auto collectors so that they can be reconfigured
+            // cleanly for each test
+            CorrelationContextManager.reset();
+            AppInsights.dispose();
+            sandbox.restore();
+        });
+
+        it("should pass along traceparent/tracestate header if present in current operation", (done) => {
+            var eventEmitter = new EventEmitter();
+            (eventEmitter as any).headers = {};
+            (eventEmitter as any)["getHeader"] = function (name: string) { return this.headers[name]; };
+            (eventEmitter as any)["setHeader"] = function (name: string, value: string) { this.headers[name] = value; };
+            (<any>eventEmitter).method = "GET";
+            sandbox.stub(https, 'request', (url: string, c: Function) => {
+                process.nextTick(c);
+                return eventEmitter;
+            });
+
+            AppInsights
+                .setup("ikey")
+                .setAutoCollectDependencies(true)
+                .start();
+
+            sandbox.stub(CorrelationContextManager, "getCurrentContext", () => ({
+                operation: {
+                    traceparent: new TraceParent("00-5e84aff3af474588a42dcbf3bd1db95f-1fc066fb77fa43a3-00"),
+                    tracestate: "sometracestate"
+                },
+                customProperties: {
+                    serializeToHeader: (): null => null
+                }
+            }));
+            https.request(<any>'https://httpbin.org/status/200', (c) => {
```

Comment:
Are these calls actually made during unit tests? Or is request mocked out here?

Reply:
I believe these calls are mocked. Will switch to url that the rest of the tests are using to avoid confusion